src/fluree/db/nameservice/proto.cljc

```diff
@@ -0,0 +1,19 @@
+(ns fluree.db.nameservice.proto)
+
+
+(defprotocol iNameService
+  (-lookup [nameservice ledger-alias] [nameservice ledger-alias opts] "Performs lookup operation on ledger alias and returns map of latest commit and other metadata")
+  (-push [nameservice commit-data] "Pushes new commit to nameservice.")
+  (-subscribe [nameservice ledger-alias callback] "Creates a subscription to nameservice(s) for ledger events")
+  (-unsubscribe [nameservice ledger-alias] "Unsubscribes to nameservice(s) for ledger events")
+  (-sync? [nameservice] "Indicates if nameservice updates should be performed synchronously, before commit is finalized. Failure will cause commit to fail")
```

Comment:
It does live on the conn. A conn can get passed nameservices, else it will default to the nameservice we think makes the most sense for the conn. I did hard-code in the `:sync?` parameter on some, but ideally a developer can construct this however they like. So what you describe is how it operates.